scripts/lib/links/FileBatch.ts

```diff
@@ -66,43 +66,36 @@ export class FileBatch {
    *   2. A list of InternalLink objects to validate.
    *   3. A list of ExternalLink objects to validate.
    */
-  async load(): Promise<[File[], InternalLink[], ExternalLink[]]> {
+  async load(
+    loadExternalLinks: boolean,
+  ): Promise<[File[], InternalLink[], ExternalLink[]]> {
     const files: File[] = [];
     for (let filePath of this.toLoad) {
       const parsed = await parseFile(filePath);
       files.push(new File(filePath, parsed.anchors));
     }
 
-    const linksToOriginFiles = new Map<string, string[]>();
+    const internalLinksToOriginFiles = new Map<string, string[]>();
+    const externalLinksToOriginFiles = new Map<string, string[]>();
     for (const filePath of this.toCheck) {
       const parsed = await parseFile(filePath);
       files.push(new File(filePath, parsed.anchors));
-      if (!IGNORED_FILES.has(filePath)) {
-        addLinksToMap(filePath, parsed.rawLinks, linksToOriginFiles);
-      }
-    }
-
-    const internalLinks: InternalLink[] = [];
-    const externalLinks: ExternalLink[] = [];
-    for (let [linkPath, originFiles] of linksToOriginFiles) {
-      if (ALWAYS_IGNORED_URLS.has(linkPath)) {
-        continue;
-      }
-      originFiles = originFiles.filter(
-        (originFile) =>
-          FILES_TO_IGNORES[originFile] == null ||
-          !FILES_TO_IGNORES[originFile].includes(linkPath),
-      );
-
-      if (originFiles.length > 0) {
-        if (linkPath.startsWith("http")) {
-          externalLinks.push(new ExternalLink(linkPath, originFiles));
-        } else {
-          internalLinks.push(new InternalLink(linkPath, originFiles));
-        }
+      addLinksToMap(filePath, parsed.internalLinks, internalLinksToOriginFiles);
+      if (loadExternalLinks) {
```

Comment:
We don't waste our time storing external links if not necessary. That should slightly decrease memory usage and improve performance.